src/components/AnimalDetails/AnimalDetails.tsx

```diff
@@ -0,0 +1,29 @@
+import { useLoaderData, useNavigate } from 'react-router-dom';
+import { Animal as AnimalType } from '../../services/api';
+import styles from './AnimalDetails.module.css';
+import { Button } from '../Button';
+
+export function AnimalDetails() {
+  const { animal } = useLoaderData() as { animal: AnimalType };
```

Comment:
I dont like this useLoaderData at all, to be honest. it's not safe at all, you use type assertion all the time. if you change the names of properties in your loader, your application will crash, but not the types in the components, but the whole point of the typescript is exactly the opposite. I never used this, but I guess what can be done here at least is to create a separate wrapper hook, which will at least regulate types by loader
```typescript
export const useMainLoaderData = () => {
  return useLoaderData() as Awaited<ReturnType<typeof animalLoader>>;
};
```
and create a new for for each new loader. Or you can play with generic types to make it universal

Reply:
Fixed